File: leap-year/LeapYearTests/YearShould.cs
using LeapYear;
using Xunit;

namespace LeapYearTests
{
    public class YearShould
    {
        [Fact]
        public void NotBeALeapYearIfNotDivisbleBy4() 
        {
            Assert.False(LeapYear(1997));
        }

        [Theory]
        [InlineData(1996)]
        [InlineData(2004)]
        public void BeALeapYearIfDivisibleBy4(int year)
        {
            Assert.True(LeapYear(year));
        }

        [Fact]
        public void BeALeapYearIfDivisibleBy400()
        {
            Assert.True(LeapYear(1600));
        }

        [Fact]
        public void NotBeALeapYearIfDivisibleBy100ButNot400()
        {
            Assert.False(LeapYear(1800));
        }

        private bool LeapYear(int year)
        {
            return new Year(year).IsLeapYear();
        }
    }
}

File: leap-year/LeapYearFSharpTests/YearShould.fs
module YearShould

open System
open Xunit
open LeapYearFSharp

[<Fact>]
let ``Not be a leap year if not divisible by 4`` () =
    Assert.False(Year.IsLeapYear 1997)
    
[<Theory>]
[<InlineData(1996)>]
[<InlineData(2004)>]
let ``Be a leap year if divisible by 4`` (year) =
    Assert.True(Year.IsLeapYear year)
    
[<Fact>]
let ``Be a leap year if divisible by 400`` () =
    Assert.True(Year.IsLeapYear 1600)
    
[<Fact>]
let ``Not be a leap year if divisible by 100 but not 400`` () =
    Assert.False(Year.IsLeapYear 1800)